internal/service/flexcluster/resource.go

```diff
@@ -106,23 +107,23 @@ func (r *rs) Read(ctx context.Context, req resource.ReadRequest, resp *resource.
 }
 
 func (r *rs) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
-	var tfModel TFModel
-	resp.Diagnostics.Append(req.Plan.Get(ctx, &tfModel)...)
+	var plan TFModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
 	if resp.Diagnostics.HasError() {
 		return
 	}
 
-	flexClusterReq, diags := NewAtlasUpdateReq(ctx, &tfModel)
+	flexClusterReq, diags := NewAtlasUpdateReq(ctx, &plan)
 	if diags.HasError() {
 		resp.Diagnostics.Append(diags...)
 		return
 	}
 
-	projectID := tfModel.ProjectId.ValueString()
-	clusterName := tfModel.Name.ValueString()
+	projectID := plan.ProjectId.ValueString()
+	clusterName := plan.Name.ValueString()
 
 	connV2 := r.Client.AtlasV2
-	_, _, err := connV2.FlexClustersApi.UpdateFlexCluster(ctx, projectID, tfModel.Name.ValueString(), flexClusterReq).Execute()
+	_, _, err := connV2.FlexClustersApi.UpdateFlexCluster(ctx, projectID, plan.Name.ValueString(), flexClusterReq).Execute()
```

Comment:
All the attributes that have the custom plan modifier in the schema are the non-updatable attributes. Having nonUpdatableStringAttributePlanModifier instead of an if in the Update operation makes this more reusable and I think will help in the auto-generation (it's easier to add a plan modifier in an attribute rather than adding custom logic in the update operation)

Reply:
discussed offline, good to go

Reply:
+1 on the automation aspect

Reply:
Curious why we don't use the `stringplanmodifier.RequiresReplace()`?